### DISCUSSION TITLE
Support for package tagging (categorization)?

### DISCUSSION THREAD

**Discussion by tddschn** [Homebrew/discussions#2089] [Everyday usage] [answered]
E.g. 

The formula `git` can have tags `["developer-tool", "version-control"]` and `tags:  ["developer-tool", "version-control"]` in the formula file, 

and users can query the registry / cloned taps for formulas that have a certain tag like this:
```bash
$ brew search-tag version-control
git
subversion
...
```
Reasons for using package tagging:
- Better package discoveribility
-

What do you think of package tagging?

Is this something the homebrew devs would consider adding support of?

**Comment by carlocab** [comment, marked as answer]
That would be nice to have, but I think that 1) adding tags to ~6000 formulae, and 2) keeping them maintained would be quite laborious. It's much more likely that most of these tags will be wrong and not useful, so I don't think this is something we'd want to implement, unfortunately.

However, if you're interested in implementing this, you can create your own [external command](https://docs.brew.sh/External-Commands) that for it. You might wish to pattern the tag database after [`command-not-found`](https://github.com/Homebrew/homebrew-command-not-found).

**Comment by SMillerDev** [comment]
If you want to send a pull request I'm sure it'll be considered. But package discovery is really not a goal of Homebrew, so I don't expect anyone else to work on this.

**Comment by tddschn** [comment]
@carlocab @SMillerDev Makes sense. Thank you for the explanations!
